wallet/src/background/Transactions.ts

```diff
@@ -73,6 +74,55 @@ class Transactions {
         );
     }
 
+    public async executeTransactionBytes(
+        txBytes: Base64DataBuffer,
+        connection: ContentScriptConnection
+    ) {
+        const txBytesRequest = this.createTransactionBytesRequest(
```

Comment:
I think we can reuse the existing `createTransactionRequest`. If you update the  tx type [here](https://github.com/MystenLabs/fastnft/blob/314e028461bf407816365d1675d30485ccf32f5d/wallet/src/shared/messaging/messages/payloads/transactions/TransactionRequest.ts) to add `Base64DataBuffer` we should be able to reuse the existing functionality and there will be no need to create new messages, `storeTransactionBytesRequest` etc

Reply:
by that do you mean adding a new optional txBytes like 
```
export type TransactionRequest = {
    id: string;
    approved: boolean | null;
    tx: MoveCallTransaction;
    txBytes?: Base64DataBuffer;
    origin: string;
    originFavIcon?: string;
    txResult?: TransactionResponse;
    txResultError?: string;
    createdDate: string;
};
```
or we want both tx and txBytes to be optional like below

```
export type TransactionRequest = {
    id: string;
    approved: boolean | null;
    tx?: MoveCallTransaction;
    txBytes?: Base64DataBuffer;
    origin: string;
    originFavIcon?: string;
    txResult?: TransactionResponse;
    txResultError?: string;
    createdDate: string;
};
```
I think with the prev one, the Dapp would have to send out unserialized txn, so it will not be useful. So following the latter one, in `respondToTransactionRequest`, we will need to tell if it is tx or txBytes and call execution method accordingly, is that also what you think? Thanks!

Reply:
I meant to do something like

```
export type TransactionRequest = {
    ...
    tx?: MoveCallTransaction | Base64DataBuffer;
    ...
};
```

and do something like `isMoveCalTransaction(req.tx)` to check if it's a move call or serialised data and call the appropriate execution method.

I think we can do something like the following that might be more clear

```
export type TransactionRequest = {
    id: string;
    approved: boolean | null;
    origin: string;
    originFavIcon?: string;
    txResult?: TransactionResponse;
    txResultError?: string;
    createdDate: string;
} & (
    {
        type: 'move-call';
        tx: MoveCallTransaction;
    } |
    {
        type: 'serialised-transaction';
        tx: string; // I think we should avoid using the Base64DataBuffer
    }
);
```

And then check the type field to chose the method to execute the transaction